hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -278,6 +319,71 @@ public HoodieWriteClient(JavaSparkContext jsc, HoodieWriteConfig clientConfig,
     }
   }
 
+  /**
+   * Loads the given HoodieRecords, as inserts into the table. This is suitable for doing big bulk
+   * loads into a Hoodie table for the very first time (e.g: converting an existing dataset to
+   * Hoodie).  The input records should contain no duplicates if needed.
+   *
+   * This implementation uses sortBy (which does range partitioning based on reservoir sampling) and
+   * attempts to control the numbers of files with less memory compared to the {@link
+   * HoodieWriteClient#insert(JavaRDD, String)}. Optionally it allows users to specify their own
+   * partitioner. If specified then it will be used for repartitioning records. See {@link
+   * UserDefinedBulkInsertPartitioner}.
+   *
+   * @param preppedRecords HoodieRecords to insert
+   * @param commitTime Commit Time handle
+   * @param bulkInsertPartitioner If specified then it will be used to partition input records
+   * before they are inserted into hoodie.
+   * @return JavaRDD[WriteStatus] - RDD of WriteStatus to inspect errors and counts
+   */
+  public JavaRDD<WriteStatus> bulkInsertPreppedRecords(JavaRDD<HoodieRecord<T>> preppedRecords,
+      final String commitTime,
+      Option<UserDefinedBulkInsertPartitioner> bulkInsertPartitioner) {
+    writeContext = metrics.getCommitCtx();
+    // Create a Hoodie table which encapsulated the commits and files visible
+    HoodieTable<T> table = HoodieTable.getHoodieTable(
+        new HoodieTableMetaClient(jsc.hadoopConfiguration(), config.getBasePath(), true),
+        config);
+    try {
+      JavaRDD<WriteStatus> writeStatusRDD = bulkInsertInternal(preppedRecords, commitTime,
+          table, bulkInsertPartitioner);
+      return updateIndexAndCommitIfNeeded(writeStatusRDD, table, commitTime);
```

Comment:
can we move `updateIndexAndCommitIfNeeded()` part also inside `bulkInsertInternal` ?